federation/planner.go

```diff
@@ -0,0 +1,300 @@
+package federation
+
+import (
+	"errors"
+	"fmt"
+	"sort"
+
+	"github.com/samsarahq/thunder/graphql"
+)
+
+type StepKind int
+
+const (
+	KindType StepKind = iota
+	KindField
+)
+
+type PathStep struct {
+	Kind StepKind
+	Name string
+}
+
+type Plan struct {
+	Path    []PathStep
+	Service string
+	Kind    string
+	// XXX: What are we using Type for here again? -- oh, it's for the __federation field...
```

Comment:
could you get rid of this comment, and document the top level types? also happy to do this in a follow-up pr.

Reply:
:+1: for documenting the public types, I think that would make this code much easier to follow.  Could I ask that we also document the fields in `Plan`?  It would be super helpful to know what kinds of values we expect for "Kind", "Type", etc. without having to refer to the test cases.